### PR TITLE
Upgrade example

### PR DESCRIPTION
**Pull request by mhevery**

### CODE REVIEW

**Comment by petebacondarwin** [comment]
I like this API much better.

It is now clear that I have an Angular 1 module, which contains Angular 1 "things" and happens to also include a "downgraded" Angular 2 component. And that this Angular 2 component expects to include an "upgraded" Angular 1 component in its template.

Nice

**Comment by mhevery** [comment]
Hi Julie,

I would like yo to have a look at this and try to make it work with Protractor E2E test. The issue is that this is a hybrid application running both Angular JS v1 and Angulra v2, and it confuses protractor.

-- @mhevery

File: modules/examples/e2e_test/upgrade/upgrade_spec.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      clickComponentButton('user', 'button');
+      expect(getComponentText('hello-app', '.greeting')).toEqual('Greetings from !');
+    });
+  });
+
+});
+
+function getComponentText(selector, innerSelector) {
+  return browser.executeScript('return document.querySelector("' + selector + '").querySelector("' +
+                               innerSelector + '").textContent');
```
Comment (gkalpak) [inline modules/examples/e2e_test/upgrade/upgrade_spec.ts]: Isn't this the same as `document.querySelector(selector + ' '  ± innerSelector)` ?
Why not use just one selector (to simplify things) ?

File: modules/examples/src/upgrade/index.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import {Component, Directive, Injectable, Input, Output, EventEmitter} from 'angular2/core';
```
Comment (gkalpak) [inline modules/examples/src/upgrade/index.ts]: `Directive`, `Injectable` aren't used.

**Comment by mhevery** [comment]
@juliemr Would you have any time to work in this?

**Comment by juliemr** [comment]
Sorry for the slow response. Yes, I can take a look (or find someone to :) )

**Comment by bradlygreen** [comment]
@juliemr are you still the owner on this?  Can you get it fixed and in this week?

**Comment by juliemr** [comment]
I don't have a good estimate for this as I haven't had a chance to look into it. I'll take a look and get an estimate by eod.

I don't think that this is a beta0 blocker, though, but maybe @mhevery can comment on that.

**Comment by mhevery** [comment]
I don't think it is beta0 blocker either.

**Comment by jelbourn** [comment]
@mhevery @juliemr what's the status of this PR? Still valid or obsolete?

**Pull request closed without merging**